crates/re_log_types/src/data_table_batcher.rs

```diff
@@ -0,0 +1,787 @@
+        Self::default().apply_env()
+    }
+
+    /// Returns a copy of `self`, overriding existing fields with values from the environment if
+    /// they are present.
+    ///
+    /// See [`Self::ENV_FLUSH_TICK`], [`Self::ENV_FLUSH_NUM_BYTES`], [`Self::ENV_FLUSH_NUM_BYTES`].
+    pub fn apply_env(&self) -> DataTableBatcherResult<Self> {
+        let mut new = self.clone();
+
+        if let Ok(s) = std::env::var(Self::ENV_FLUSH_TICK) {
+            let flush_duration_secs: f64 =
+                s.parse()
+                    .map_err(|err| DataTableBatcherError::ParseConfig {
+                        name: Self::ENV_FLUSH_TICK,
+                        value: s.clone(),
+                        err: Box::new(err),
+                    })?;
+
+            new.flush_tick = Duration::from_secs_f64(flush_duration_secs);
+        }
+
+        if let Ok(s) = std::env::var(Self::ENV_FLUSH_NUM_BYTES) {
+            new.flush_num_bytes = s
+                .parse()
+                .map_err(|err| DataTableBatcherError::ParseConfig {
+                    name: Self::ENV_FLUSH_NUM_BYTES,
+                    value: s.clone(),
+                    err: Box::new(err),
+                })?;
+        }
+
+        if let Ok(s) = std::env::var(Self::ENV_FLUSH_NUM_ROWS) {
+            new.flush_num_rows = s
+                .parse()
+                .map_err(|err| DataTableBatcherError::ParseConfig {
+                    name: Self::ENV_FLUSH_NUM_ROWS,
+                    value: s.clone(),
+                    err: Box::new(err),
+                })?;
+        }
+
+        Ok(new)
+    }
+}
+
+#[test]
+fn data_table_batcher_config() {
+    // Detect breaking changes in our environment variables.
+    std::env::set_var("RERUN_FLUSH_TICK_SECS", "0.3");
+    std::env::set_var("RERUN_FLUSH_NUM_BYTES", "42");
+    std::env::set_var("RERUN_FLUSH_NUM_ROWS", "666");
+
+    let config = DataTableBatcherConfig::from_env().unwrap();
+
+    let expected = DataTableBatcherConfig {
+        flush_tick: Duration::from_millis(300),
+        flush_num_bytes: 42,
+        flush_num_rows: 666,
+        ..Default::default()
+    };
+
+    assert_eq!(expected, config);
+}
+
+// ---
+
+/// Implements an asynchronous batcher that coalesces [`DataRow`]s into [`DataTable`]s based upon
+/// the thresholds defined in the associated [`DataTableBatcherConfig`].
+///
+/// ## Multithreading and ordering
+///
+/// [`DataTableBatcher`] can be cheaply clone and used freely across any number of threads.
+///
+/// Internally, all operations are linearized into a pipeline:
+/// - All operations sent by a given thread will take effect in the same exact order as that
+///   thread originally sent them in, from its point of view.
+/// - There isn't any well defined global order across multiple threads.
+///
+/// This means that e.g. flushing the pipeline ([`Self::flush_blocking`]) guarantees that all
+/// previous data sent by the calling thread has been batched; no more, no less.
```

Comment:
> has been batched; no more, no less.

I think should be something like "batched *and* sent into the tx_tables crossbeam channel.